src/distributions/weighted.rs

```diff
@@ -228,3 +698,40 @@ impl fmt::Display for WeightedError {
         write!(f, "{}", self.msg())
     }
 }
+
+/// Error type returned from [`AliasMethodWeightedIndex::new`].
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum AliasMethodWeightedIndexError {
```

Comment:
Is there much justification for having a separate error type? It adds a significant amount of code for little gain.

I don't have a problem with modifying the members of `WeightedError`, though of course such a change can't land in a patch release.

Reply:
I don't think so. In the first version of the pull request the error types were a little bit more different. Now they are pretty similar. Let me know if you want to make that breaking change.

Reply:
Yes, I would prefer to do it this way. I won't be able to merge yet with a breaking change, but I'm not sure there will be more releases for 0.6.x anyway so it likely won't make much difference to when it gets released.

Reply:
Didn't we already merge some breaking change to master?

Reply:
I will change this once there is a plan for the other discussion, because it might affect this.

Reply:
I changed this in 5b2934122f04e1ff41024b3c0597ace711d21f08.